homeassistant/components/onkyo/media_player.py

```diff
@@ -159,351 +149,191 @@ def service_handle(service):
-                    receiver,
-                    config.get(CONF_SOURCES),
-                    name=config.get(CONF_NAME),
-                    max_volume=config.get(CONF_MAX_VOLUME),
-                    receiver_max_volume=config.get(CONF_RECEIVER_MAX_VOLUME),
-                )
-            )
-            KNOWN_HOSTS.append(host)
-
-            zones = determine_zones(receiver)
-
-            # Add Zone2 if available
-            if zones["zone2"]:
-                _LOGGER.debug("Setting up zone 2")
-                hosts.append(
-                    OnkyoDeviceZone(
-                        "2",
-                        receiver,
-                        config.get(CONF_SOURCES),
-                        name=f"{config[CONF_NAME]} Zone 2",
-                        max_volume=config.get(CONF_MAX_VOLUME),
-                        receiver_max_volume=config.get(CONF_RECEIVER_MAX_VOLUME),
-                    )
-                )
-            # Add Zone3 if available
-            if zones["zone3"]:
-                _LOGGER.debug("Setting up zone 3")
-                hosts.append(
-                    OnkyoDeviceZone(
-                        "3",
-                        receiver,
-                        config.get(CONF_SOURCES),
-                        name=f"{config[CONF_NAME]} Zone 3",
-                        max_volume=config.get(CONF_MAX_VOLUME),
-                        receiver_max_volume=config.get(CONF_RECEIVER_MAX_VOLUME),
-                    )
-                )
-        except OSError:
-            _LOGGER.error("Unable to connect to receiver at %s", host)
-    else:
-        for receiver in eISCP.discover():
-            if receiver.host not in KNOWN_HOSTS:
-                hosts.append(OnkyoDevice(receiver, config.get(CONF_SOURCES)))
-                KNOWN_HOSTS.append(receiver.host)
-    add_entities(hosts, True)
-
-
-class OnkyoDevice(MediaPlayerDevice):
-    """Representation of an Onkyo device."""
-
-    def __init__(
-        self,
-        receiver,
-        sources,
-        name=None,
-        max_volume=SUPPORTED_MAX_VOLUME,
-        receiver_max_volume=DEFAULT_RECEIVER_MAX_VOLUME,
-    ):
-        """Initialize the Onkyo Receiver."""
-        self._receiver = receiver
-        self._muted = False
-        self._volume = 0
-        self._pwstate = STATE_OFF
-        self._name = (
-            name or f"{receiver.info['model_name']}_{receiver.info['identifier']}"
+    _LOGGER.debug("Provisioning Onkyo AVR device at %s:%d", host, port)
+
+    @callback
+    def async_onkyo_update_callback(message):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.debug("Received update callback from AVR: %s", message)
+        zone, _, _ = message
+        if zone in active_zones.keys():
+            active_zones[zone].process_update(message)
+            hass.async_create_task(active_zones[zone].async_update_ha_state())
+
+    avr = await pyeiscp.Connection.create(
```

Comment:
Please wrap it in a try / catch and raise `PlatformNotReady` if it fails

`from homeassistant.exceptions import PlatformNotReady`